org.eclipse.xtext.ide.tests/src/org/eclipse/xtext/ide/tests/server/concurrent/RequestManagerTest.xtend

```diff
@@ -42,6 +46,35 @@ class RequestManagerTest {
 		sharedState = null
 	}
 
+	@Test(timeout = 1000)
+	def void testLogException() {
+		val logResult = LoggingTester.captureLogging(Level.ALL, RequestManager, [
+			val future = requestManager.runWrite([
+				throw new RuntimeException();
+			], [])
+			
+			// join future to assert log later
+			try {
+				future.get
+			} catch (Exception e) {}
+		])
+		
+		logResult.assertLogEntry("Error during request:")
+	}
+
+	@Test(timeout = 1000, expected = ExecutionException)
+	def void testCatchException() {
+		LoggingTester.captureLogging(Level.ALL, RequestManager, [
+			val future = requestManager.runWrite([
+				throw new RuntimeException()
+			], [])
+
+			assertEquals('Foo', future.get)
+		])
+		
+		Assert.fail
```

Comment:
This last `Assert.fail` is not necessary

Reply:
I know. I added that to highlight that the test case will not reach that point, but instead end with the thrown exception at `future.get`. You want that to be removed?

Reply:
I'm ok with it being there. Maybe `Assert.fail("unreachable")`